src/exceptions.cpp

```diff
@@ -2,6 +2,9 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#ifndef ASTARTE_EXCEPTIONS_HPP
```

Comment:
This is used in headers to avoid double inclusions. It makes no sense to have it here. What was your objective?